### ISSUE TITLE
`TypeVar` cannot handle `TypeVar("Name", *types)`

### ISSUE THREAD

**Issue by rentruewang** [microsoft/pyright#2215]
**Describe the bug**

Suppose that `types` is a sequence of types, for example `(int, float)`,
then `TypeVar("Name", *types)` should be exactly the same as `TypeVar("Name", int, float)`.

However, `TypeVar("Name", *types)* yields errors.

**To Reproduce**
See the below **Screenshots or Code** section.

**Expected behavior**
Both should pass the checker.

**Screenshots or Code**

The following two files should be completely equivilant.

file1.py:

```python
from typing import TypeVar

Real = TypeVar("Real", int, float)

def p(x: Real) -> None:
    print(x)

p(int(3))
p(float(4))
```

`file2.py`:

```python
from typing import TypeVar

types = (int, float)
Real = TypeVar("Real", *types)

def p(x: Real) -> None:
    print(x)

p(int(3))
p(float(4))
```

For `file1.py`, the check passes.
For `file2.py`, the check fails with:
```shell
/home/me/Documents/file2.py
  /home/me/Documents/file2.py:4:25 - error: Expected class type but received "tuple[Type[int], Type[float]]" (reportGeneralTypeIssues)
  /home/me/Documents/file2.py:4:25 - error: TypeVar cannot have only a single constraint (reportGeneralTypeIssues)
  /home/me/Documents/file2.py:9:3 - error: Argument of type "int" cannot be assigned to parameter "x" of type "Real@p" in function "p"
    Type "int" is incompatible with constrained type variable "Real" (reportGeneralTypeIssues)
  /home/me/Documents/file2.py:10:3 - error: Argument of type "float" cannot be assigned to parameter "x" of type "Real@p" in function "p"
    Type "float" is incompatible with constrained type variable "Real" (reportGeneralTypeIssues)
4 errors, 0 warnings, 0 infos
```

**VS Code extension or command-line**
Command line tool, version: `pyright 1.1.162`.

**Additional context**
Tested on both Linux and Mac. Both produce the same issue.

**Comment by erictraut** [comment]
This is intentional. You're using a complex, dynamic expression as an argument to TypeVar, but static type checkers expect these arguments to be simple type expressions — either a class name or a type alias that refers to a class.

Other Python type checkers have this same restriction. Mypy, for example, emits the same error.

The workaround is to not use a complex expression (e.g. an unpack operator) to specify the arguments to `TypeVar`.
